client/mksubvolume.cc

```diff
@@ -326,45 +396,84 @@ doit()
     // Determine name for new subvolume: It is the target name without the
     // leading filesystem target.
 
-    string subvolume_name = target.substr(fs_target.size() + (fs_target == "/" ? 0 : 1));
+    string subvolume_name = target.substr(fs_target.size() +
+					 (fs_target == "/" || fs_target == target ? 0 : 1));
     if (verbose)
 	cout << "subvolume-name:" << subvolume_name << endl;
 
-    // Execute all steps.
+    // Set up cache for UUID / LABEL resolution in mnt_table_find_source
+    libmnt_cache* cache = mnt_new_cache();
+    libmnt_table* mtab_table = mnt_new_table();
+    mnt_table_set_cache(mtab_table, cache);
+    mnt_table_parse_mtab(mtab_table, NULL);
+
+    char* subvol_expected;
+    libmnt_fs* expected_fs = create_fstab_line(fs, subvol_option, subvolume_name);
+    mnt_fs_get_option(expected_fs, "subvol", &subvol_expected, NULL);
+    assert(subvol_expected != NULL);
+
+    // Consistency checks on (partially) existing entries
+    libmnt_fs* fstab_entry = mnt_table.find_target(target, MNT_ITER_FORWARD);
```

Comment:
Below are two tests for fstab_entry != NULL. Why not throw an exception right here in that case?